File: mvp_compat_views/src/dagger/java/com/mandria/android/mvp/provider/PresenterClass.java
package com.mandria.android.mvp.provider;

import com.mandria.android.mvp.Presenter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * This annotation is used to ensure only presenters can be provided by the {@link PresenterFactory}.
 */
@Inherited
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface PresenterClass {

    /**
     * @return Class type of a presenter class.
     */
    Class<? extends Presenter> value();
}

File: views_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/provider/PresenterClass.kt
package com.mandria.android.mvp.provider

import com.mandria.android.mvp.Presenter
import java.lang.annotation.Inherited
import kotlin.reflect.KClass

/** Annotation to specify presenter classes to use as key with dagger binding. **/
@Inherited
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class PresenterClass(val value: KClass<out Presenter<*>>)